cmd/stringer/golden_test.go

```diff
@@ -296,11 +327,222 @@ func (i Token) String() string {
 }
 `
 
+// Simple test: enumeration of type int starting at 0.
+const month_in = `type Month int
+const (
+	January Month = iota
+	February
+	March
+	April
+	May
+	June
+	July
+	August
+	September
+	October
+	November
+	December
+)
+`
+
+const month_out = `
+import "strconv"
+
+const _Month_name = "JanuaryFebruaryMarchAprilMayJuneJulyAugustSeptemberOctoberNovemberDecember"
+
+var _Month_index = [...]uint8{0, 7, 15, 20, 25, 28, 32, 36, 42, 51, 58, 66, 74}
+
+func (i Month) String() string {
+	if i < 0 || i >= Month(len(_Month_index)-1) {
+		return "Month(" + strconv.FormatInt(int64(i), 10) + ")"
+	}
+	return _Month_name[_Month_index[i]:_Month_index[i+1]]
+}
+
+func MonthFromString(s string) (i Month, ok bool) {
+	for v := range _Month_index {
+		v += 0
+		if s == Month(v).String() {
+			return Month(v), true
+		}
+	}
+	return
+}
+`
+
+// Enumeration with an negative offset.
+const mood_in = `type Mood int
+const (
+	Negative Mood = iota -1
+	Neutral
+	Happy
+)
+`
+
+const mood_out = `
+import "strconv"
+
+const _Mood_name = "NegativeNeutralHappy"
+
+var _Mood_index = [...]uint8{0, 8, 15, 20}
+
+func (i Mood) String() string {
+	i -= -1
+	if i < 0 || i >= Mood(len(_Mood_index)-1) {
+		return "Mood(" + strconv.FormatInt(int64(i+-1), 10) + ")"
+	}
+	return _Mood_name[_Mood_index[i]:_Mood_index[i+1]]
+}
+
+func MoodFromString(s string) (i Mood, ok bool) {
+	for v := range _Mood_index {
+		v += -1
+		if s == Mood(v).String() {
+			return Mood(v), true
+		}
+	}
+	return
+}
+`
+
+// Gaps and an offset with duplicates.
+const fib_in = `type Fib int
+const (
+	Zero       Fib = 0
+	One        Fib = 1
+	AnotherOne Fib = 1
+	Two        Fib = 2
+	Three      Fib = 3
+	Five       Fib = 5
+	Eight      Fib = 8
+	Thirteen   Fib = 13
+)
+`
+
+const fib_out = `
```

Comment:
warning: don't use underscores in Go names; const fib_out should be fibOut (golint)